packages/react-component-event-listener/src/lib/shouldUpdateListener.ts

```diff
@@ -0,0 +1,8 @@
+import { EventListenerProps } from '../types'
+
+const shouldUpdateListener = (prevProps: EventListenerProps, nextProps: EventListenerProps) =>
```

Comment:
actually, the real intent we would like to express here is that "do not update if only `listener` was changed". Thus, lets make this intent explicit by providing a logic like the following instead:

```jsx
const shouldUpdateListener = (prevProps: EventListenerProps, nextProps: EventListenerProps) =>
  if (shallowEqual(
       { ...prevProps, listener: nextProps.listener },
       nextProps)) 
    {
       return false 
    } 
   ...
```

This will save us from any kind of maintainability problems that might lead to bugs that would be hard to discover - like, for example, the ones that will appear in case we would need to extend the props of this component

Reply:
I support this idea, however I should mention that in this case we will add a new dependency for short term... As we are going to update to React 16.8, may be we can go with existing implementation? _However, I am fully agree that your proposal is more understandable..._

Reply:
agree with your point - lets discuss in person and decide? or, if you feel strongly about merging, please, feel free to skip this comment :) in that case we will be able to handle this in a follow-up PR

Reply:
Agreed to `shallowEqual()` function to this file and remove it fully when we will migrate to hooks